fix for ticket:13301: Removed redundant UVector entry clean up function call.
from https://ssl.icu-project.org/trac/changeset/40324/trunk/icu4c/source/i18n/zonemeta.cpp
Index: trunk/icu4c/source/i18n/zonemeta.cpp
===================================================================
--- a/icu/source/i18n/zonemeta.cpp	(revision 40283)
+++ b/icu/source/i18n/zonemeta.cpp	(revision 40324)
@@ -693,5 +693,4 @@
                     if (U_FAILURE(status)) {
                         delete mzMappings;
-                        deleteOlsonToMetaMappingEntry(entry);
                         uprv_free(entry);
                         break;
